cmake/config_posix.h.cmake

```diff
@@ -0,0 +1,12 @@
+
+#ifndef __CONFIG_POSIX_H__
+#define __CONFIG_POSIX_H__
+
+#cmakedefine _XOPEN_SOURCE @_XOPEN_SOURCE@
+
+/* to get htobe16, le16toh, etc */
+#define _BSD_SOURCE
+#include <endian.h>
```

Comment:
Unfortunately, this fails on my computer (running FreeBSD):

```
nfc-tools/libfreefare/build % make
Scanning dependencies of target freefare
[  2%] Building C object libfreefare/CMakeFiles/freefare.dir/felica.c.o
In file included from /home/romain/Projects/nfc-tools/libfreefare/libfreefare/felica.c:28:
/home/romain/Projects/nfc-tools/libfreefare/config.h:9:10: fatal error: 'endian.h' file not found
#include <endian.h>
         ^~~~~~~~~~
1 error generated.
*** Error code 1
```

I guess it will fail on Mac OS too…  Can't we detect `sys/endian.h`, `endian.h`, `CoreFoundation/CoreFoundation.h` and `byteswap.h` and define `HAVE_SYS_ENDIAN_H`, `HAVE_ENDIAN_H`, `HAVE_COREFOUNDATION_COREFOUNDATION_H` and `HAVE_BYTESWAP_H` accordingly?